go/vt/vttablet/tabletserver/tabletserver.go

```diff
@@ -513,32 +513,28 @@ func (tsv *TabletServer) fullStart() (err error) {
 	}
 	tsv.hr.Init(tsv.target)
 	tsv.updateStreamList.Init()
+	tsv.te.Open()
 	return tsv.serveNewType()
 }
 
 func (tsv *TabletServer) serveNewType() (err error) {
+	if !tsv.te.isOpen {
```

Comment:
I actually put this in here to make `cache_invalidation.py` green again. Not sure what is going on there, but if I comment these three lines out, that test starts failing.

Reply:
It just occurred to me that cache_invalidation did expose a problem with the approach of opening `te` in `fullStart`. Essentially, we have to close and re-open `te` every time `serveNewType` gets called. This is because `Close` will have to wait for existing transactions to be resolved before we can switch to the new serving type.

In other words, we should go back to the old behavior, except that we immediately `te.Open` as soon as it's closed.